server/src/main/java/io/crate/planner/consumer/InsertFromSubQueryPlanner.java

```diff
@@ -64,7 +64,7 @@ public static LogicalPlan plan(AnalyzedInsertStatement statement,
         }
 
         List<Reference> targetColsExclPartitionCols = new ArrayList<>(
-            statement.columns().size() - statement.tableInfo().partitionedBy().size());
+            Math.max(1, statement.columns().size() - statement.tableInfo().partitionedBy().size()));
```

Comment:
This is only changing the initial capacity of the list, how is this fixing a bug?

Reply:
Nevermind, it became negative before.

Reply:
yep